pvactools/lib/top_score_filter.py

```diff
@@ -215,28 +228,14 @@ def find_best_line(self, lines):
         sorted_anchor_residue_pass_lines = sorted(anchor_residue_pass_lines, key=lambda d: (float(d["{} MT IC50 Score".format(self.mt_top_score_metric)]), d['TSL Sort'], -int(d['Transcript Length'])))
         return sorted_anchor_residue_pass_lines[0]
 
-    def get_anchor_positions(self, hla_allele, epitope_length):
-        if self.allele_specific_anchors and epitope_length in self.anchor_probabilities and hla_allele in self.anchor_probabilities[epitope_length]:
-            probs = self.anchor_probabilities[epitope_length][hla_allele]
-            positions = []
-            total_prob = 0
-            for (pos, prob) in sorted(probs.items(), key=lambda x: x[1], reverse=True):
-                total_prob += float(prob)
-                positions.append(pos)
-                if total_prob > self.anchor_contribution_threshold:
-                    return positions
-
-        return [1, 2, epitope_length - 1 , epitope_length]
-
-
     def is_anchor_residue_pass(self, line):
         if self.use_allele_specific_binding_thresholds:
             binding_threshold = self.allele_specific_binding_thresholds[line['HLA Allele']]
         else:
             binding_threshold = self.binding_threshold
 
         anchor_residue_pass = True
-        anchors = self.get_anchor_positions(line['HLA Allele'], len(line['MT Epitope Seq']))
+        anchors = get_anchor_positions(self, line['HLA Allele'], len(line['MT Epitope Seq']))
```

Comment:
I don't think we'll want to pass `self` here. In order to make this generic and future-proof, e.g. for other classes where the attributes might be named differently, we should pass along all attributes we use from `self` explicitly as their own parameters.